src/currentprocess/filesource.rs

```diff
@@ -193,14 +185,36 @@ impl Isatty for TestWriter {
     }
 }
 
+impl Terminal for TestWriter {
+    fn fg(&mut self, _: crate::cli::term::Color) -> io::Result<()> {
+        Ok(())
+    }
+
+    fn bg(&mut self, _: crate::cli::term::Color) -> io::Result<()> {
+        Ok(())
+    }
+
+    fn attr(&mut self, _: crate::cli::term::Attr) -> io::Result<()> {
+        Ok(())
+    }
+
+    fn reset(&mut self) -> io::Result<()> {
+        Ok(())
+    }
+
+    fn carriage_return(&mut self) -> io::Result<()> {
+        self.0.lock().unwrap().write(b"\r").map(|_| ())
```

Comment:
`Ok(self.0.lock()?.write(b"\r")?;())`

might be a little better. Or not - whatever you like.

Reply:
Updated!

Reply:
Updated!